### PR TITLE
Updating the Radius installation step of the long-haul workflow

### PR DESCRIPTION
**Pull request by ytimocin**
# Description

* Added **Report Failure** step to the workflow.
* Removed the secret deletion workaround step.

Details can be found here: #7007 

## Type of change
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

### CODE REVIEW

File: .github/workflows/long-running-azure.yaml [hunk 1]
```diff
@@ -349,6 +352,14 @@ jobs:
...
         env:
           RESOURCE_GROUP: ${{ env.AZURE_TEST_RESOURCE_GROUP }}
+      - name: Check if tests are already running
+        run: |
+          if kubectl get configmap long-running-test-lock -n radius-system --ignore-not-found; then
+            echo "Tests are already running. Exiting..."
+            exit 1
+          fi
+      - name: Set test run flag
+        run: kubectl create configmap long-running-test-lock -n radius-system
```
Comment (youngbupark) [inline .github/workflows/long-running-azure.yaml:362]: What if the clean up step in previous workflow run is failed? the current run will be ignored.

File: .github/workflows/long-running-azure.yaml [hunk 2]
```diff
@@ -368,6 +379,12 @@ jobs:
           export PATH=$GITHUB_WORKSPACE/bin:$PATH
           which rad || { echo "cannot find rad"; exit 1; }
 
+          echo "*** Uninstalling existing Radius installation ***"
+          rad uninstall kubernetes
+
+          echo "*** Deleting radius-system namespace ***"
+          kubectl delete namespace radius-system --ignore-not-found
```
Comment (youngbupark) [inline .github/workflows/long-running-azure.yaml:386]: I do not think this is right approach. it is still workaround. upgrade scenario should work here.

File: deploy/Chart/templates/controller/webhook.yaml [hunk 3]
```diff
@@ -32,7 +33,7 @@ webhooks:
 - admissionReviewVersions:
   - v1
   clientConfig:
-    caBundle: {{ b64enc $ca.Cert }}
+    caBundle: {{ $validatingWebhookCaBundle }}
```
Comment (ytimocin) [inline deploy/Chart/templates/controller/webhook.yaml:36]: The missing piece was that we were always using a new caBundle here. If the secret stays the same and pod gets recreated, we shouldn't create a new caBundle for this. That creates all the failures of second time test runs.

Also, ucp does it this way: https://github.com/radius-project/radius/blob/main/deploy/Chart/templates/ucp/apiservice.yaml#L42.

cc/ @youngbupark @rynowak @lakshmimsft

Comment (youngbupark) [reply deploy/Chart/templates/controller/webhook.yaml:36]: Can you please open new PR for helm update? this is the actual fix for upgrading scenario. So it is better to create new pr for this.

File: deploy/Chart/templates/controller/validating-webhook-configuration.yaml [hunk 4]
```diff
@@ -13,7 +13,6 @@ metadata:
   labels:
     app.kubernetes.io/name: controller
     app.kubernetes.io/part-of: radius
-type: kubernetes.io/tls
```
[the remark this answers is not in the document]
Comment (ytimocin) [reply deploy/Chart/templates/controller/validating-webhook-configuration.yaml:16]: Created this issue: https://github.com/radius-project/radius/issues/7035.

**Review by lakshmimsft — APPROVED** [review]

**Comment by lakshmimsft** [comment]
as discussed, secrets continue to be 'opaque' as they were earlier.

**Pull request merged**